datadog/resource_datadog_dashboard.go

```diff
@@ -4601,6 +4852,158 @@ func getTimeseriesRequestSchema() map[string]*schema.Schema {
 		},
 	}
 }
+
+func buildDatadogFormula(data map[string]interface{}) datadogV1.WidgetFormula {
+	formula := datadogV1.WidgetFormula{}
+	if formulaExpression, ok := data["formula_expression"].(string); ok && len(formulaExpression) != 0 {
+		formula.SetFormula(formulaExpression)
+	}
+	if alias, ok := data["alias"].(string); ok && len(alias) != 0 {
+		formula.SetAlias(alias)
+	}
+	if limits, ok := data["limit"].([]interface{}); ok && len(limits) != 0 {
+		datadogLimit := datadogV1.NewWidgetFormulaLimit()
+		limit := limits[0].(map[string]interface{})
+		if count, ok := limit["count"].(int); ok && count != 0 {
+			datadogLimit.SetCount(int64(count))
+		}
+		if order, ok := limit["order"].(string); ok && len(order) > 0 {
+			datadogLimit.SetOrder(datadogV1.QuerySortOrder(order))
+		}
+		formula.SetLimit(*datadogLimit)
+	}
+	return formula
+}
+
+func buildDatadogEventQuery(data map[string]interface{}) datadogV1.FormulaAndFunctionQueryDefinition {
+	dataSource := datadogV1.FormulaAndFunctionEventsDataSource(data["data_source"].(string))
+	computeList := data["compute"].([]interface{})
+	computeMap := computeList[0].(map[string]interface{})
+	aggregation := datadogV1.FormulaAndFunctionEventAggregation(computeMap["aggregation"].(string))
+	compute := datadogV1.NewTimeSeriesFormulaAndFunctionEventQueryDefinitionCompute(aggregation)
+	if interval, ok := computeMap["interval"]; ok && interval.(int) != 0 {
+		compute.SetInterval(int64(interval.(int)))
+	}
```

Comment:
We spent a little time offline looking into how to make sure we're safely accessing these. Since we found that checking the `ok` when retrieving the key + checking the lengh/value is enough, can we do the type assertion in the if:

```suggestion
	if interval, ok := computeMap["interval"].(int); ok && interval != 0 {
		compute.SetInterval(int64(interval))
	}
```

Same for the few following (the rest of the PR seems to follow this pattern too ^

Reply:
Addressed in commit [5c22cd7](https://github.com/DataDog/terraform-provider-datadog/pull/892/commits/5c22cd771533e25ac58a9e87bc9da024672d34f3).